data/src/server.rs

```diff
@@ -72,7 +116,7 @@ impl<'a> From<(&'a Server, &'a Arc<config::Server>)> for Entry {
 }
 
 #[derive(Debug, Clone, Default, Deserialize)]
-pub struct Map(IndexMap<Server, Arc<config::Server>>);
+pub struct ConfigMap(IndexMap<ServerName, Arc<config::Server>>);
```

Comment:
Not sure we need both `Map` and `ConfigMap`. A `Server` can be created from a bare `ServerName`, which is valid to use on `Config.servers`

Reply:
<img width="913" height="997" alt="Screenshot_20250731-000001" src="https://github.com/user-attachments/assets/a8f92c24-3b4d-4731-b3bc-ca6e50d604cf" />
<img width="938" height="817" alt="Screenshot_20250731-000047" src="https://github.com/user-attachments/assets/6068696c-7c9e-43e2-bee0-996100d19ee0" />

this is why i designed it this way